### DISCUSSION TITLE
Course Progress is not working in NIC env

### DISCUSSION THREAD

**Discussion by somvitbhowmik** [Sunbird-Lern/Community#40] [Query]
Hi Tean,

I have one course which has MP4 PDF and assessment types of resources. All the resources are 100% complete and status is 2 in user-content-consumption table. But in user-entollments table data is not updated. Status is still 1.  Status should be 2 once all the resources are 100% complete. But it is not updated. Pls help us on this. @reshmi-nair @maheshkumargangula

cc: @pkranga

Regards,
Somvit

**Comment by pkranga** [comment]
@reshmi-nair there are not error logs in the activity aggregator 

Also is there a process to run the enrolment-reconciliation flink job. I am assuming that the job reconcilles all the progress and updates the progress completion.

**Reply by reshmi-nair** [reply]
@somvitbhowmik , @pkranga  Open the course from UI and please choose the option "Sync progress now" from 3-dots menu next to Course Progress section to sync the progress of the course. Also pls check all the flink jobs [Assessment-aggregator, activity aggregator, enrolment reconciliation, relation cache updater etc.] status to see whether any other issue is there or not.

**Reply by pkranga** [reply]
I dont see the option to sync from UI. Also please note this is in iGOT for which we do not use the Sunbird ED UI. The backend services are directly deployed from sunbird repo though
![Screenshot from 2022-09-20 19-06-38](https://user-images.githubusercontent.com/46988344/191272554-376188b0-d77f-4fa9-9ddf-bbbb664da484.png)

**Reply by pkranga** [reply]
@reshmi-nair

**Reply by pkranga** [reply]
@reshmi-nair @maheshkumargangula any updates?

**Reply by chitranshu-keshav** [reply]
@rhwarrier @reshmi-nair can you please get back on this

**Reply by reshmi-nair** [reply]
@somvitbhowmik Let's meet in a call and discuss this

**Reply by somvitbhowmik** [reply]
@reshmi-nair We have identified the root cause of this issue. Issue is in redis cache. hierarchy data and leafnodes data are not matching. But count is same. Scenario is :

1. Create a course do review and publish
2. Verify in redis hierarchy data and $courseId:$courseId:leafNodes data
3. Edit the course and remove one resource and add new resource. Then review and publish.
4. Consume the course you will see that course get stuck in 85%.
5. Now trigger the jenkins job Neo4jElasticSearchSyncTool and select "sync-collection-rel-cache" command 
6. verify the data  in redis hierarchy data and $courseId:$courseId:leafNodes data. Compare the leafnodes of hierarchy with leafNodes.

cc: @pkranga

**Reply by reshmi-nair** [reply]
@somvitbhowmik We are testing the above scenario from our end. I will update the thread soon.

**Reply by prasadmoka** [reply]
@somvitbhowmik  can you please share both the events that are being pushed to kafka topic:"sunbirddev.content.postpublish.request" for each publish. As per the above scenario, we should have 2 events pushed to kafka for the relation-cache-updator to process these 2 events. 
Meantime we are checking the same in our local environment as well.

**Reply by pkranga** [reply]
@prasadmoka were you able to replicate the scenario in your local env?

**Reply by prasadmoka** [reply]
@pkranga @somvitbhowmik we have asked somvit to share the kafka events as we should have 2 kafka events pushed, so that relation cache updater can process them. In our dev and staging environment it was working fine. Can we have a call with anyone from your team tomorrow

**Reply by pkranga** [reply]
@somvitbhowmik pl set up a call tomorrow to show the issue in our environment

**Reply by somvitbhowmik** [reply]
@prasadmoka We will schedule a call on monday. Hope it is fine for you. Pls share your email as well.

**Reply by prasadmoka** [reply]
@somvitbhowmik pls send invite to prasad@sanketika.in
